src/coreclr/src/vm/marshalnative.cpp

```diff
@@ -277,7 +277,7 @@ FCIMPL2(UINT32, MarshalNative::SizeOfClass, ReflectClassBaseObject* refClassUNSA
     // refClass is validated to be non-NULL RuntimeType by callers
     TypeHandle th = refClass->GetType();
 
-    if (throwIfNotMarshalable && !th.IsBlittable())
+    if (throwIfNotMarshalable && (!th.IsBlittable() || th.IsArray()))
```

Comment:
Aside based on previous conversation.

Does the `char` type return true for `IsBlittable`? I am trying to understand what happens here for

```CSharp
struct A
{
   char c;
}
```

Reply:
Based on the surrounding code am I to assume that the struct marshallers handle the above type and treat `char` as a blittable type?

Reply:
Yes, the above structure is handled by the struct marshalling infra (the `GetNativeSize` call below).